spec/classes/init_spec.rb

```diff
@@ -273,6 +273,35 @@
 
   describe 'config files' do
     platforms.sort.each do |k,v|
+      context "when configuring pam_d_sshd_template on #{v[:osfamily]} with #{v[:releasetype]} #{v[:release]}" do
+        let :facts do
+          { :osfamily => v[:osfamily],
+            :"#{v[:releasetype]}" => v[:release],
+            :lsbdistid => v[:lsbdistid],
+          }
+        end
+
+        let (:params) do
+          { :pam_d_sshd_template     => 'pam/sshd.custom.erb',
+            :pam_sshd_auth_lines     => [ 'auth_lines' ],
+            :pam_sshd_account_lines  => [ 'account_lines' ],
+            :pam_sshd_session_lines  => [ 'session_lines ' ],
+            :pam_sshd_password_lines => [ 'password_lines' ],
+          }
+        end
+
+        if v[:osfamily] != 'Solaris'
```

Comment:
Sure thing. The reason for doing this way in the first place was that I don't quite like conditionals in tests and thought this would be a tad more readable and obvious. Not sure what I think any more so I'll change it ;)